app/controllers/users_controller.rb

```diff
@@ -1,20 +1,36 @@
+# frozen_string_literal: true
+
 class UsersController < ApplicationController
+  before_action :set_user, only: %i[edit update]
+
   def new
-    @r = User.new
+    @user = User.new
   end
 
   def create
     @r = User.create(user_params)
 
-   if @r
-    redirect_to new_user_path
-   else
-    render :new
-   end
+    if @r
+      redirect_to new_user_path
+    else
+      render :new
+    end
+  end
+
+  def update
+    @user.update(user_params)
```

Comment:
- [ ] you should have two cases for this method; when the user gets updated and when it doesn't.

Reply:
now when the user is not updated we will be take to users/index